plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py

```diff
@@ -119,6 +119,13 @@ def launch(
             for x, y in params.items()
             if x not in init_keys
         }
+        # submitit `setup` parameter must be a primitive list container
+        if f"{self._EXECUTOR}_setup" in params and isinstance(
+            params[f"{self._EXECUTOR}_setup"], ListConfig
+        ):
+            params[f"{self._EXECUTOR}_setup"] = OmegaConf.to_container(
```

Comment:
what happens when setup = None?

also why do you need to check if its ListConfig? is it to get past mypy?

Reply:
Won't the if statement be false if it's None?  

I've been reading through the code trying to get a better handle of the assumptions for initializing the Launcher.  The code doesn't indicate that the input for initialization comes from configuration objects, can I assume this? If so, in the `__init__` function could I just do:

```python
self.params["setup"] = OmegaConf.to_container(self.params["setup"], resolve=True)
```

Reply:
A simpler approach is to convert all the params to primitive containers.
This seems to work:
```diff
diff --git a/plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py b/plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py
index 1b3635c7e..b862ab5b1 100644
--- a/plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py
+++ b/plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py
@@ -114,12 +114,17 @@ class BaseSubmititLauncher(Launcher):
 
         # specify resources/parameters
         baseparams = set(OmegaConf.structured(BaseQueueConf).keys())
-        params = {
-            x if x in baseparams else f"{self._EXECUTOR}_{x}": y
-            for x, y in params.items()
-            if x not in init_keys
-        }
-        executor.update_parameters(**params)
+
+        params = OmegaConf.create(
+            {
+                x if x in baseparams else f"{self._EXECUTOR}_{x}": y
+                for x, y in params.items()
+                if x not in init_keys
+            }
+        )
+
+        plain_params = OmegaConf.to_container(params, resolve=True)
+        executor.update_parameters(**plain_params)
 
         log.info(

```

You can also convert them inside the existing overly fancy dict comprehension loop (slightly more efficient but much more convoluted, at least in the current form of things.

Reply:
You can also just convert all the OmegaConf containers in params to containers.